src/dmd/todt.d

```diff
@@ -789,6 +789,9 @@ private void membersToDt(AggregateDeclaration ad, ref DtBuilder dtb,
     uint offset;
     if (cd)
     {
+        const bool gentypeinfo = global.params.useTypeInfo && Type.dtypeinfo;
+        const bool genclassinfo = gentypeinfo || !(cd.isCPPclass || cd.isCOMclass);
```

Comment:
Note that we do need ClassInfos for `extern(C++)` classes etc. too, e.g., to use them as AA keys/values. If they aren't emitted anymore (I can't really tell - DMD glue...) but the testsuite still passes, that'd be quite horrible. This should still compile & link:
```D
extern(C++) class C {}

void main()
{
    C[int] aa = [ 0: new C() ];
    import core.stdc.stdio;
    printf("size: %d\n", cast(int) typeid(C).initializer.length);
}
```

Reply:
Ah wait, so you always emit for non-betterC *and* D classes (even with `-betterC`).

Reply:
IIRC, LDC never emits any TypeInfo with `!global.params.useTypeInfo`; to make `extern(D)` classes somewhat usable with `-betterC` too, the first ptr in the vtable (pointer to dynamic type's ClassInfo) is null in that case (so dynamic casts and dynamic `typeid` won't work etc.).

Reply:
> Ah wait, so you always emit for non-betterC or D classes (even with -betterC).

Yes. I didn't modify the current logic, I just went along with the current implementation and added the same check where it made sense.

Is there any change that you are asking for this PR?

Reply:
I guess no, not related to the `extern(C++)` issue at hand.